apps/microtvm/reference-vm/base-box-tool.py

```diff
@@ -293,19 +318,34 @@ def do_run_release_test(release_test_dir, provider_name, test_config, test_devic
         os.path.join(release_test_dir, ".vagrant", "machines", "default", provider_name, "id")
     ) as f:
         machine_uuid = f.read()
-    ATTACH_USB_DEVICE[provider_name](
-        machine_uuid,
-        vid_hex=test_config["vid_hex"],
-        pid_hex=test_config["pid_hex"],
-        serial=test_device_serial,
-    )
+
+    # Check if target is not QEMU
+    if test_config["vid_hex"] and test_config["pid_hex"]:
+        ATTACH_USB_DEVICE[provider_name](
+            machine_uuid,
+            vid_hex=test_config["vid_hex"],
+            pid_hex=test_config["pid_hex"],
+            serial=test_device_serial,
+        )
     tvm_home = os.path.realpath(os.path.join(THIS_DIR, "..", "..", ".."))
 
     def _quote_cmd(cmd):
         return " ".join(shlex.quote(a) for a in cmd)
 
-    test_cmd = _quote_cmd(["cd", tvm_home]) + " && " + _quote_cmd(test_config["test_cmd"])
-    subprocess.check_call(["vagrant", "ssh", "-c", f"bash -ec '{test_cmd}'"], cwd=release_test_dir)
+    test_cmd_zephyr = (
+        _quote_cmd(["cd", tvm_home]) + " && " + _quote_cmd(test_config["test_cmd_zephyr"])
+    )
+    test_cmd_zephyr_aot = (
+        _quote_cmd(["cd", tvm_home]) + " && " + _quote_cmd(test_config["test_cmd_zephyr_aot"])
+    )
+    if test_cmd_zephyr:
```

Comment:
rather than complicate this logic with one-invocation-per-test-command, i was thinking we may add a shell script here and just invoke that as `test_cmd`. wdyt?

Reply:
that's a good suggestion. I changed it run a single script that has all tests and get microtvm-platform as input target.